### ISSUE TITLE
Keyboard display issues

### ISSUE THREAD

**Issue by Heineken** [rdoeffinger/Dictionary#33]
On my device I often find the keyboard behaviour irritating:

- While typing I realize that I am in the wrong language. I switch and the keyboard disappears although I am not done typing. (reproducible)
- On the other hand sometimes the keyboard unnecessarily pops up over dictionary list hiding the dictionaries in the lower part (not reproducible)

**Comment by rdoeffinger** [comment]
On the first one: On the other hand, for people who finish typing and then switch the language the current behaviour is more convenient.
What do you think about making it dependent on whether an exact match for the word is found?
It won't be 100% (for example short words likely would have a match that isn't the word meant, so maybe there should be a minimum length). I could imagine it to either be quite convenient or really annoying because of the inconsistency...

If you have any additional information on the second one, like in which circumstances it happens most, that would be nice to know.

**Comment by Heineken** [comment]
Hmm, I disagree on the first one. I usually don't type out the whole word, only as much as necessary to make the desired word show up. That's why I switch the language midway. For that reason your fix wouldn't help until I get used to typing more.

Also making the keyboard disappear is less necessary (I can still see the result) and quicker (slow phone) than making it reappear.

I'll keep an eye on the 2nd one.

**Comment by rdoeffinger** [comment]
> That's why I switch the language midway. For that reason your fix wouldn't help until I get used to typing more.

Sorry? My proposal would NOT hide the keyboard UNLESS whatever you typed matches a word exactly.
So if you typed "tes" it would keep the keyboard, if it was "test" it would hide it.
That said you're probably right it might be better to just keep it.

**Comment by Heineken** [comment]
Sorry, I didn't read your proposal carefully.

But still, I want to look up "testicles" and suddenly, while typing, the keyboard disappears... which makes me tap something unintentionally. (OK, not a problem here, since tapping an entry doesn't do anything, but this is my favourite. In the old QuickDict. the dictionary list often jumped around just when I tapped a flag, which made me tap the wrong flag. Good riddance this was fixed.)
